test/e2e/tests/setup.ts

```diff
@@ -26,25 +22,26 @@ export const setup = async (): Promise<SetupResult> => {
     return cachedSetup;
   }
 
-  const publicClient = setupPublicClient();
   const engine = setupEngine();
   const backendWallet = await getEngineBackendWallet(engine);
-  let testClient;
+  await engine.backendWallet.resetNonces();
 
   if (CONFIG.USE_LOCAL_CHAIN) {
-    setupAnvil();
+    startAnvil();
```

Comment:
if we `await startAnvil()` we can probably get rid of the `sleep(1000)`, I assume prool helps us know when anvil has started producing blocks?

Reply:
Yep good point, will add as a follow up